src/app/user-profile-search/user-profile-filter/user-profile-filter.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { UserProfileSearchResult } from '../user-profile-search-result';
+import { FilterValue } from '../filter-value';
+import { FilterName } from './filter-name.enum';
+
+@Component({
+  selector: 'app-user-profile-filter',
+  templateUrl: './user-profile-filter.component.html',
+  styleUrls: ['./user-profile-filter.component.scss']
+})
+export class UserProfileFilterComponent implements OnInit {
+
+  @Input() foundUsers: UserProfileSearchResult[];
+  public filteredSearchResult: UserProfileSearchResult[];
+
+  public certificatesFilter: FilterValue[];
+  public skillsFilter: FilterValue[];
+  public industrySectorsFilter: FilterValue[];
+  public positionProfilesFilter: FilterValue[];
+
+  constructor() {
+  }
+
+  ngOnInit() {
+    this.filteredSearchResult = this.foundUsers;
+
+    this.certificatesFilter = this.getCertificatesValues();
+    this.skillsFilter = this.getSkillsValues();
+    this.industrySectorsFilter = this.getIndustrySectorsValues();
+    this.positionProfilesFilter = this.getPositionProfileValues();
+  }
+
+  filter() {
+    this.filteredSearchResult = this.foundUsers.filter(user =>
```

Comment:
I was not aware of this possible solution. 😄 We should really try it with the client-side filtering approach first. We will have to transfer more user data as part of the search API response but then we can do all the filtering without any further server request. Thinking about the expected number of search results in our business scenario this should be OK. - Great solution! 👍